tests/runner.py

```diff
@@ -216,8 +216,8 @@ def build(self, src, dirname, filename, output_processor=None, main_file=None, a
                ['-I', dirname, '-I', os.path.join(dirname, 'include')] + \
                map(lambda include: '-I' + include, includes) + \
                ['-c', f, '-o', f + '.o']
-        output = Popen(args, stdout=PIPE, stderr=self.stderr_redirect if not DEBUG else None).communicate()[0]
-        assert os.path.exists(f + '.o'), 'Source compilation error: ' + output
+        output = subprocess.check_call(args, stdout=PIPE, stderr=self.stderr_redirect if not DEBUG else None)
```

Comment:
Reading a second pass, I notice that here and below, `check_call` is used with `PIPE`. That is not allowed, since it is subject to a deadlock: https://docs.python.org/2/library/subprocess.html .  (I think this is dumb from python and python should throw an exception instead if `check_call` is used with `PIPE`. They are practically documenting their bugs as misfeatures.) Popen should be used here instead.